core/arch/arm/plat-stm32mp1/conf.mk

```diff
@@ -42,6 +42,9 @@ ifeq ($(flavor_dts_file-$(PLATFORM_FLAVOR)),)
 $(error Invalid platform flavor $(PLATFORM_FLAVOR))
 endif
 CFG_EMBED_DTB_SOURCE_FILE ?= $(flavor_dts_file-$(PLATFORM_FLAVOR))
+else
+PLATFORM_FLAVOR ?= 157C_DK2
+CFG_EMBED_DTB_SOURCE_FILE ?= $(flavor_dts_file-$(PLATFORM_FLAVOR))
```

Comment:
Simpler:
```patch
+PLATFORM_FLAVOR ?= 157C_DK2
-ifneq ($(PLATFORM_FLAVOR),)
 ifeq ($(flavor_dts_file-$(PLATFORM_FLAVOR)),)
 $(error Invalid platform flavor $(PLATFORM_FLAVOR))
 endif
 CFG_EMBED_DTB_SOURCE_FILE ?= $(flavor_dts_file-$(PLATFORM_FLAVOR))
-endif
```

That said, build directive could config for, for example, a MP15-EV1 board wihtout setting `PLATFORM_FLAVOR`, for example `make CFG_EMBED_DTB_SOURCE_FILE=stm32mp157c-ev1.dts`
or for an MP13 variant: `make CFG_EMBED_DTB_SOURCE_FILE=stm32mp135f-dk.dts CFG_STM32MP13=y`

In those case, having `PLATFORM_FLAVOR=157C_DK2` would have not effect but would be confusing (it would appears in the generate config files, next to non 157c-dk2 configs). Therefore, I would prefer we set a default embedded DTB if none is provided, rather than forcing a `PLATFORM_CONFIG` value:
```patch
 ifneq ($(PLATFORM_FLAVOR),)
 ifeq ($(flavor_dts_file-$(PLATFORM_FLAVOR)),)
 $(error Invalid platform flavor $(PLATFORM_FLAVOR))
 endif
 CFG_EMBED_DTB_SOURCE_FILE ?= $(flavor_dts_file-$(PLATFORM_FLAVOR))
 endif
+CFG_EMBED_DTB_SOURCE_FILE ?= stm32mp157c-dk2.dts
```

What do you think?

Reply:
I'm ok with your suggestion.

Only thing is we can have a flavor different from the embedded device tree. But that was already the case so... :)

I will rebusmit next week